libbpf-rs/tests/test.rs

```diff
@@ -242,3 +242,57 @@ fn test_object_link_pin() {
     link.unpin().expect("failed to unpin prog");
     assert!(!Path::new(path).exists());
 }
+
+#[test]
+fn test_object_reuse_pined_map() {
+    bump_rlimit_mlock();
+
+    let path = "/sys/fs/bpf/mymap";
+    let key = vec![1, 2, 3, 4];
+    let val = vec![1, 2, 3, 4, 5, 6, 7, 8];
+
+    // Pin a map
+    {
+        let mut obj = get_test_object();
+        let map = obj
+            .map("start")
+            .expect("error finding map")
+            .expect("failed to find map");
+
+        map.update(&key, &val, MapFlags::empty())
+            .expect("failed to write");
+
+        // Pin map
+        map.pin(path).expect("failed to pin map");
+        assert!(Path::new(path).exists());
+    }
+
+    // Reuse the pinned map
+    let obj_path = get_test_object_path();
+    let mut builder = ObjectBuilder::default();
+    builder.debug(true);
+    let mut open_obj = builder.open_file(obj_path).expect("failed to open object");
+
+    let start = open_obj
+        .map("start")
+        .expect("error finding map")
+        .expect("failed to find map");
+    assert!(start.reuse_map("/asdf").is_err());
```

Comment:
will assert!() panic? if yes, you won't unpin the map, wouldn't that be a problem?

Reply:
Good point; I'll fix it